test/server_integration/http/platform/status.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { ServiceStatus, ServiceStatusLevels } from '../../../../src/core/server';
+import { FtrProviderContext } from '../../services/types';
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  const delay = (ms: number) => new Promise((r) => setTimeout(r, ms));
+  const getStatus = async (pluginName: string): Promise<ServiceStatus> => {
+    const resp = await supertest.get('/api/status?v8format=true');
+
+    return resp.body.status.plugins[pluginName];
+  };
+
+  const setStatus = async <T extends keyof typeof ServiceStatusLevels>(level: T) =>
+    supertest
+      .post(`/internal/core_plugin_a/status/set?level=${level}`)
+      .set('kbn-xsrf', 'xxx')
+      .expect(200);
+
+  describe('status service', () => {
+    // This test must comes first because the timeout only applies to the initial emission
+    it("returns a timeout for status check that doesn't emit after 30s", async () => {
+      let aStatus = await getStatus('corePluginA');
+      expect(aStatus.level).to.eql('unavailable');
+
+      // Status will remain in unavailable due to core services until custom status timesout
+      // Keep polling until that condition ends, up to a timeout
+      const start = Date.now();
+      while (aStatus.summary !== 'Status check timed out after 30s') {
```

Comment:
Is there a reason not to use `retry.waitForWithTimeout` from the standard library?